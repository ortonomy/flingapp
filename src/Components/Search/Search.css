.Search .search-box {
  width: 100%;
  border-bottom: solid 1px black;
}

.Search .search-box input {
  border: none;
  background-color: inherit;
  width: 100%;
  padding: 0.75em 0;
}

.Search .search-box input::placeholder {
    color: rgba(0,0,0,0.4);
}

.Search .search-box input:focus {
  box-shadow: none;
  outline-width: 0;
}
