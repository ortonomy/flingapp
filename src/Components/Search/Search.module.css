.Search {
  composes: paddedBox from '../Sidebar/Sidebar.module.css';
}

.box {
  width: 100%;
  border-bottom: solid 1px black;
}

.box input {
  border: none;
  background-color: inherit;
  width: 100%;
  padding: 0.75em 0;
}

.box input::placeholder {
    color: rgba(0,0,0,0.4);
}

.box input:focus {
  box-shadow: none;
  outline-width: 0;
}
