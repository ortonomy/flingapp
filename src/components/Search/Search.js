import React, { Component } from 'react';
import search from './Search.module.css';
import sidebar from '../Sidebar/Sidebar.module.css';

class Search extends Component {

  search() {
    return true;
  }

  render() {
    return (
      <div className={search.Search}>
        <div className={sidebar.sectionTitle}>
          <i className='fa fa-search'></i>
          <span>Search</span>
        </div>
        <div className={search.box}>
          <input type="text" onClick={this.search} placeholder="find projects and freelancers" />
        </div>
      </div>
    )
  }
}

export default Search;
