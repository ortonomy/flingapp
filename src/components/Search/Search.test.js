import React from 'react';
import Search from '../Search';

it('renders without crashing', () => {
  const wrapper = shallow(<Search />);
  expect(wrapper).toMatchSnapshot();
});
