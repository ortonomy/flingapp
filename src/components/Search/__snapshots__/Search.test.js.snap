// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className={undefined}
>
          <i
                    className="fa fa-search"
          />
          <span>
                    Search
          </span>
</div>,
        <div
          className={undefined}
>
          <input
                    onClick={[Function]}
                    placeholder="find projects and freelancers"
                    type="text"
          />
</div>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="fa fa-search"
/>,
            <span>
              Search
</span>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-search",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search",
            },
            "ref": null,
            "rendered": "Search",
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            onClick={[Function]}
            placeholder="find projects and freelancers"
            type="text"
/>,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onClick": [Function],
            "placeholder": "find projects and freelancers",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className={undefined}
>
            <i
                        className="fa fa-search"
            />
            <span>
                        Search
            </span>
</div>,
          <div
            className={undefined}
>
            <input
                        onClick={[Function]}
                        placeholder="find projects and freelancers"
                        type="text"
            />
</div>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="fa fa-search"
/>,
              <span>
                Search
</span>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fa fa-search",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Search",
              },
              "ref": null,
              "rendered": "Search",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              onClick={[Function]}
              placeholder="find projects and freelancers"
              type="text"
/>,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onClick": [Function],
              "placeholder": "find projects and freelancers",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
